hal/inc/cellular_hal.h

```diff
@@ -138,6 +138,13 @@ cellular_result_t cellular_global_identity(CellularGlobalIdentity* cgi, void* re
 
 bool cellular_sim_ready(void* reserved);
 
+/**
+ * Change the timeout a SIM has to register on the cellular network before the modem is reset
+ * 
+ * @param timeout: registration timeout in milliseconds
+ */
+cellular_result_t cellular_registration_timeout_set(system_tick_t timeout, void* reserved = NULL);
```

Comment:
This is a C header and a C function, there cannot be a default argument, which is a C++-specific feature.

Reply:
Good catch

Reply:
Modular and monolithic builds will need separate calls.  One with a nullptr and the other without.

Reply:
Why? It's a standard practice in DeviceOS to leave a `void* reserved` for ABI compatibility and future extensibility and to require the caller to fill that with `nullptr` by default. What is the difference between monolithic and modular builds in this regard?

Reply:
Fixed